### PR TITLE
Add segmentation blog

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
@gwaybio 

This blog PR is ready for review! :confetti_ball:

### CODE REVIEW

File: _posts/2022-10-31-segmentation.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+
+{%
+  include figure.html
+  image="images/blog/segmentation/sensitivity_blog_fig.png"
+  width="100%"
+%}
+
+> Figure 8. Comparison of sensitivity between Cellpose and CellProfiler. This figure shows how CellProfiler segments cells much tighter and includes small parts compared to Cellpose. Cellpose segments cells more broadly and includes more of the background compared to CellProfiler.
+
+After doing a lot of prototyping with both CellProfiler and Cellpose, I believe that Cellpose segments whole cells much better (e.g. no cell is segmented over each other) than CellProfiler (Figure 9).
```
Comment (gwaybio) [inline _posts/2022-10-31-segmentation.md:272]: wait a sec, I thought you just said in lines 261 and 262:

> One of the biggest differences I noticed between Cellpose and CellProfiler was that Cellpose segmentations did not fit the shape of the cells as much as CellProfiler. 
> As well, CellProfiler included more of the stringy parts of the actin that Cellpose did not (Figure 8).

Doesn't this contradict that Cellpose segments whole cells much better? Where am I confused?

File: _posts/2022-10-31-segmentation.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+This will impact downstream processing and analysis since you won't be able to, for example, connect the cells to their nuclei without additional customization (e.g. create custom Python functions, etc.).
+
+In all, I think that the CellProfiler Cellpose plugin would work much better if it included the CellProfiler method of using segmented nuclei as the base for segmenting the whole cells or to add functionality that can connect the cytoplasm to their respective nucleus (or nuclei). 
+This will be a project that I will be pursuing in the future!
+
+## Conclusion
+
+After all of the prototyping, I have come to one conclusion: All these segmentation methods are good! 
+They are all very easy to use and seem to work well by eye. 
+It is yet to be seen if these subtle segmentation differences that I discussed will impact cell morphologies and thesubsequent biology that we discover (e.g. biomarkers for gene deficiency).
```
Comment (gwaybio) [inline _posts/2022-10-31-segmentation.md:300]:
```suggestion
It is yet to be seen if these subtle segmentation differences that I discussed will impact cell morphologies and the subsequent biology that we discover (e.g. biomarkers for gene deficiency).
```

File: _posts/2022-10-31-segmentation.md [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+I hope that my testing has helped you and I will be seeing you next time when I go over feature extraction methods!
+
+{%
+  include figure.html
+  image="images/blog/segmentation/segmentation_table.png"
+  width="100%"
+%}
+
+> Table 1. Pros and cons between segmentation methods.
```
Comment (gwaybio) [inline _posts/2022-10-31-segmentation.md:314]: Do I recall that you said Cellpose does in fact have a CLI?

Comment (gwaybio) [reply _posts/2022-10-31-segmentation.md:314]: I think python should be row 3 (before prototyping) (super nitpicky, I know)

**Review by gwaybio — CHANGES_REQUESTED** [review]
Looks great! I made a couple comments, mostly minor. I'll take one last look prior to approving and then you can go ahead and merge!

File: _posts/2022-10-31-segmentation.md [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+2.  Cellpose
+
+3.  Cellpose plugin for CellProfiler
+
+I also tested Ilastik, Weka Trainable Segmentation, and Scikit testing.
+From my experience, these performed sub-optimally with my data. I will discuss these methods in part II of my segmentation blog.
+
+**Note:** My segmentation findings in this blog are based on Cell Painting data. 
+For more information on Cell Painting assays and what makes them unique, go through the [GitHub wiki](https://github.com/carpenterlab/2022_Cimini_NatureProtocols/wiki#morphological-image-feature-extraction-from-microscopy-data) from the Broad Institute.
```
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md:30]: 👍

File: _posts/2022-10-31-segmentation.md [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+Cell diameter is so important because if set too large, it could over-segment objects (e.g. include background or merge cells). 
+But set too small, it will incorrectly segment a bunch of small regions within the object instead of segmenting it whole. (Figure 3)
+
+{%
+  include figure.html
+  image="images/blog/segmentation/cell_diameter_blog.png"
+  width="100%"
+%}
+
+> Figure 3. Comparison of cell diameter (CD) impact on segmentation. This figure demonstrates how CD values can make a big impact on how Cellpose segments nuclei. The left panel shows what happens when the CD is very low (segmenting nuclei into multiple parts) and the right panel shows what happens when the CD is very high (combining nuclei with abstracts). CD was the most important parameter that I toggled when optimizing nuclei segmentation.
```
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md:143]: Tbh I wasn't really sure what they are. I assumed they were abstracts but definitely could be parts of the cell. I can reword this.

File: _posts/2022-10-31-segmentation.md [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+%}
+
+> Figure 5. Cellpose 2.0 parameters available in the CellProfiler Cellpose plugin. This figure demonstrates that the most up-to-date plugin file does work in CellProfiler and provides parameters (YAY!).
+
+Since I had so many issues with the install, and when you look up "Cellpose plugin for CellProfiler" on Google, it pulls up the forum post
+that does not give complete instructions, I have gone ahead and created [a version-controlled document](https://github.com/WayScience/CellProfiler_Prototyping/tree/main/Cellpose_Plugin_CellProfiler_Instructions) for how to install this plugin.
+
+Now, let's talk about some positives!
+
+The one thing I really liked about this plugin is that it has a parameter called `Supply nuclei image as well?'.
```
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md:217]: 👍

File: _posts/2022-10-31-segmentation.md (hunk 1 above)
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md:272]: @gwaybio 

From my perspective, I am comparing two different metrics; sensitivity and whole cell segmentation. 

CellProfiler is much more sensitive, but Cellpose segments the whole cell more accurately/better.

File: _posts/2022-10-31-segmentation.md (hunk 2 above)
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md:300]: 👍

File: _posts/2022-10-31-segmentation.md (hunk 3 above)
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md:314]: @gwaybio 

For the first comment, yes it does have a CLI but I never tested it, so I put in the key for N/A that is means "not applicable, did not test". I am sure where I could put that information.

No worries on the second comment, I will edit it.

**Comment by jenna-tomkinson** [comment]
@gwaybio 

Ready for re-review! :smile:

File: _posts/2022-11-01-segmentation.md [hunk 7]
```diff
@@ -0,0 +1,314 @@
+---
+title: "Segmentation Part I"
```
Comment (gwaybio) [inline _posts/2022-11-01-segmentation.md:2]:
```suggestion
title: "Segmentation software usability and performance: Part I"
```

Segmentation Part I is not very descriptive. Please update my title suggestion if you'd like!

File: _posts/2022-11-01-segmentation.md [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+author: Jenna Tomkinson
+member: jenna-tomkinson
+tags:
+  - opinions
+  - process
+---
+
+# A Brief Performance Comparison of Popular Cell Segmentation Software
```
Comment (gwaybio) [inline _posts/2022-11-01-segmentation.md:10]:
```suggestion
# A Brief Comparison of Popular Cell Segmentation Software
```

If you clarify "software usability and performance" in the title, then you can drop the word "performance" in the leading H1 heading.

File: _posts/2022-10-31-segmentation.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (gwaybio) [reply _posts/2022-10-31-segmentation.md]: What is an abstract?

File: _posts/2022-10-31-segmentation.md (hunk 1 above)
Comment (gwaybio) [reply _posts/2022-10-31-segmentation.md]: got it!

**Review by gwaybio — APPROVED** [review]
Looks great! I think we can merge and tweet about this today

File: _posts/2022-11-01-segmentation.md (hunk 7 above)
Comment (jenna-tomkinson) [reply _posts/2022-11-01-segmentation.md:2]: Agreed! I like this title and "usability" was totally the word I was looking for but could never place!

File: _posts/2022-11-01-segmentation.md (hunk 8 above)
Comment (jenna-tomkinson) [reply _posts/2022-11-01-segmentation.md:10]: 👍

File: _posts/2022-10-31-segmentation.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jenna-tomkinson) [reply _posts/2022-10-31-segmentation.md]: @gwaybio 

Sorry I meant artifacts LOL

**Pull request merged**